File: android/http-core-java/src/main/java/com/caldremch/http/core/abs/IConvertStrategy.java
package com.caldremch.http.core.abs;

import com.caldremch.http.core.framework.base.IBaseResp;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.InputStream;

/**
 * Created by Leon on 2022/10/30.
 */
public interface IConvertStrategy {
    <T> boolean isStreamConvert(@NotNull Class<T> cls);
    <T> T convertStream(@Nullable InputStream inputStream);
    <T> IBaseResp<T> convertCommon(@NotNull String str, @NotNull Class<T> cls);
}

File: android/http-coroutine/bin/main/com/caldremch/http/impl/IConvertStrategy.kt
package com.caldremch.http.impl

import java.io.InputStream

/**
 * Created by Leon on 2022/7/5
 */
interface IConvertStrategy{
    fun <T> isStreamConvert(clz: Class<T>):Boolean
    fun <T> convertStream(inputStream: InputStream?):T
    fun <T> convertCommon(bodyString:String, clz: Class<T>):T
}